plugins/CuraEngineBackend/StartSliceJob.py

```diff
@@ -108,7 +117,7 @@ def run(self):
         with self._scene.getSceneLock():
             # Remove old layer data.
             for node in DepthFirstIterator(self._scene.getRoot()):
-                if node.callDecoration("getLayerData"):
+                if node.callDecoration("getLayerData") and node.callDecoration("getBuildPlateNumber") == self._build_plate_number:
```

Comment:
What happens if this new decorator doesn't exist on a node and we try to compare it to a value?